app/javascript/components/widgets/guests/carousel.jsx

```diff
@@ -0,0 +1,132 @@
+const ErrorMessage = ({ message }) => <p>Error: {message}</p>;
+ErrorMessage.propTypes = {
+  message: PropTypes.string.isRequired,
+};
+
+const subscribeAnnouncementPublished = gql`
+  subscription onAnnouncementPublished {
+    announcementPublished {
+      message
+      people
+      company
+      publishOn
+      id
+    }
+  }
+`;
+
+const getPrimaryLocationAnnouncements = gql`
+  {
+    primaryLocation {
+      dayAnnouncements {
+        message
+        people
+        company
+        publishOn
+        id
+      }
+    }
+  }
+`;
+
+const AnnouncementTime = styled.div``;
+
+const AnnouncementTitle = styled.div``;
+
+const AnnouncementWrapper = styled.div`
+  display: flex;
+  height: 5vh;
+  padding: 1vh;
+`;
+
+const AnnouncementIcon = styled.div`
+  background: url(${icon});
+  width: 3vw;
+  background-repeat: no-repeat;
+`;
+
+class SubscribedAnnouncements extends React.Component {
+  componentDidMount() {
+    this.props.subscribeToPublishedAnnouncements();
+  }
+
+  render() {
+    const { dayAnnouncements } = this.props;
+    return (
+      <div>
+        {dayAnnouncements.map(item => (
+          <AnnouncementWrapper key={item.id}>
+            <AnnouncementIcon />
+            <div>
+              <AnnouncementTime>{parseTime(item.publishOn)}</AnnouncementTime>
+              <AnnouncementTitle>{item.people}</AnnouncementTitle>
+            </div>
+          </AnnouncementWrapper>
+        ))}
+      </div>
+    );
+  }
+}
+SubscribedAnnouncements.propTypes = {
+  dayAnnouncements: PropTypes.objectOf.isRequired,
+  subscribeToPublishedAnnouncements: PropTypes.func.isRequired,
+};
+
+const Guests = () => (
+  <Query query={getPrimaryLocationAnnouncements}>
+    {({ loading, error, data, subscribeToMore }) => {
+      if (loading) {
+        return <LoadingMessage />;
+      }
+
+      if (error) {
+        return <ErrorMessage message={error.message} />;
+      }
+
+      return (
+        <SubscribedAnnouncements
+          dayAnnouncements={data.primaryLocation.dayAnnouncements}
+          subscribeToPublishedAnnouncements={() =>
+            subscribeToMore({
+              document: subscribeAnnouncementPublished,
+              updateQuery: (prev, { subscriptionData }) => {
+                if (!subscriptionData.data) {
+                  return prev;
+                }
+                const { announcementPublished } = subscriptionData.data;
+
+                let result = prev;
+
+                if (isToday(announcementPublished.publishOn)) {
+                  const count = prev.primaryLocation.dayAnnouncements.length;
+                  result = assocPath(
+                    ['primaryLocation', 'dayAnnouncements', count],
+                    announcementPublished,
+                    result,
+                  );
+                  result.primaryLocation.dayAnnouncements = sortBy(
+                    prop('publishOn'),
+                    result.primaryLocation.dayAnnouncements,
+                  );
```

Comment:
Hah! That's a weird quirk of assocPath. This would be a good opportunity to compose ramda functions into something like `insertSortedBy(element, prop)(collection)` and then use `over` / `lensPath` to update inside the object store.

These slides seem like a decent overview of curry/compose in ramdajs:
http://slides.com/andydrew/ramda-js-curry-compose#/

The promise stuff is good to know but really relevant here.

We can go into more detail if you have more questions

Reply:
Decided to sort when mapping through the array of announcements and generating the components. The subscription now only needs to append the `announcementPublished` to `prev`.